dpbench/benchmarks/rodinia/gaussian/gaussian_sycl_native_ext/gaussian_sycl/_gaussian_sycl.cpp

```diff
@@ -0,0 +1,131 @@
+// SPDX-FileCopyrightText: 2022 - 2023 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "_gaussian_kernel.hpp"
+#include <CL/sycl.hpp>
+#include <dpctl4pybind11.hpp>
+
+#define BLOCK_size_XY 4
+
+template <typename... Args> bool ensure_compatibility(const Args &...args)
+{
+    std::vector<dpctl::tensor::usm_ndarray> arrays = {args...};
+
+    auto arr = arrays.at(0);
+    auto q = arr.get_queue();
+    auto type_flag = arr.get_typenum();
+    auto arr_size = arr.get_size();
+
+    for (auto &arr : arrays) {
+        if (!(arr.get_flags() & (USM_ARRAY_C_CONTIGUOUS))) {
+            std::cerr << "All arrays need to be C contiguous.\n";
+            return false;
+        }
+        if (arr.get_typenum() != type_flag) {
+            std::cerr << "All arrays should be of same elemental type.\n";
+            return false;
+        }
+        if (arr.get_ndim() > 1) {
+            std::cerr << "All arrays expected to be single-dimensional.\n";
+            return false;
+        }
+    }
+    return true;
+}
+
+void gaussian_sync(dpctl::tensor::usm_ndarray a,
+                   dpctl::tensor::usm_ndarray b,
+                   dpctl::tensor::usm_ndarray m,
+                   int size,
+                   dpctl::tensor::usm_ndarray result)
+{
+    if (!ensure_compatibility(a, m, b, result))
+        throw std::runtime_error("Input arrays are not acceptable.");
+
+    int t;
+
+    sycl::queue q_ct1;
+
+    int block_size, grid_size;
+    block_size = q_ct1.get_device()
+                     .get_info<cl::sycl::info::device::max_work_group_size>();
```

Comment:
`block_size` in dpex_k is assigned the value of `size`. But here it takes the `max_work_group_size`.

Reply:
Added dpctl implementation for getting `max_work_group_size` inside numba_dpex_k implementation.